test/modules/generic-mqtt-tester/src/tester.rs

```diff
@@ -293,22 +293,31 @@ impl MessageTester {
 fn message_handler(
     settings: &Settings,
     publish_handle: PublishHandle,
-) -> Option<Box<dyn MessageHandler + Send>> {
-    let tracking_id = settings.tracking_id();
+) -> Result<Option<Box<dyn MessageHandler + Send>>, MessageTesterError> {
```

Comment:
This logic with the `Result<Option<>>` method in tester.rs seems a bit off to me.. I don't think it is worth improving since it is a test module, but if this were a prod module I would want to make this change:

I am going through each component, checking the settings, and maybe constructing it if appropriate. I think it would be more readable if I matched on the setting type, then constructed all components. I didn't do this at first as I thought it would lead to code duplication. But there are mitigations to that, and I think the current state is too confusing.